spec/unit/facter/consul_version_spec.rb

```diff
@@ -0,0 +1,24 @@
+require "spec_helper"
+
+describe Facter::Util::Fact do
+  before {
+    Facter.clear
+  }
+
+  describe "consul_version" do
+
+    context 'Returns consul version on Linux'
+    it do
+      consul_version_output = <<-EOS
+Consul v0.6.0
+Consul Protocol: 3 (Understands back to: 1)
+      EOS
+      allow(Facter.fact(:kernel)).to receive(:value).and_return("Linux")
+      allow(Facter::Util::Resolution).to receive(:exec).with('consul --version').
+        and_return(consul_version_output)
+      expect(Facter.fact(:consul_version).value).to match(/\d+\.\d+\.\d+/)
```

Comment:
This is fine, but can you add a really strong expect fact to eql '0.6.0' so I don't have to trust the regex logic? With such a controlled input, we should be able to say *exactly* what the output was, and not worry about spaces or anything extra.